### DISCUSSION TITLE
Scripts Funktionen in html nutzen?

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#1966] [Q&A] [answered]
In einem HTML kann man anscheinend keinen getState oder subscribeState durchführen?
Versucht man es erscheint:
![image](https://user-images.githubusercontent.com/84818934/195127109-49132bd4-0747-4fba-8a94-e4bc45d31f8d.png)
Hatte es im HTML versucht:
```
htmlCode += '   <script>';
htmlCode += "       window.Socket.getState('virtualpowermeter.0.group_Buero.Virtual_Energy_Power_Laptop', ({ err, state }) => {";
htmlCode += '           console.log("GetState: "+state.val);';
htmlCode += '      });';
htmlCode += '   </script>';
```

Wo liegt der Fehler?

Wie kann ich dem html Code eine Variable vom `scripts` übergeben?

**Comment by Zefau** [comment, marked as answer]
Nein, das Scripting ist isoliert. Du kannst den Wert höchsten im Scripting laden und dann im HTML verwenden.

Also im Scriping `<div id="meinWert">XXX</div>` und im Scriping im `getState` den Wert setzen `document.getElementById('meinWert').innerHTML = 'der Wert';`
